Add xpatch implementation and move to cmake

rodarima opened the pull request.

rodarima committed:
Add xpatch implementation and move to cmake

trufae commented:
im not a big fan of cmake (actually i hate it) xD but i'm fine as long as it doesnt affect r2. i added basic support for this format in r2 in rapatch and radiff2, but i didnt added any test yet and there are many ways to improve it to not just create smarter patches but also to support more formats in rapatch. my plan is to reuse the testcases you publish here for the r2 testsuite, and maybe at some point when a new release of r2 is out we can have a ci job that tests your thing with r2 in this repo too

trufae commented:
so i fine if you merge this pr, lets move things forward!

rodarima committed:
Allow override of PATH for xdiff and xpatch

Makes it possible to test other implementations of xdiff and xpatch
against the test suite. Adds an example for radiff2 (from radare2) and
can be tested with:

	XPATCH_PATH=/path/to/xpatch/test/r2 ctest

Missing binaries will fall back to the ones provided by xpatch.

rodarima committed:
Accept filenames without metadata

rodarima committed:
Remove tee command from test
